### PR TITLE
[subdevice] items_optional for Extended color light

### PR DESCRIPTION
**Pull request by Zehir**
Part 13 of https://github.com/dresden-elektronik/deconz-rest-plugin/pull/7155

### CODE REVIEW

**Review by ebaauw — CHANGES_REQUESTED** [review]
See also https://github.com/dresden-elektronik/deconz-rest-plugin/wiki/Mandatory-and-Optional-Items-for-Lights-Resources.

I do think we might have a couple of lights that don't support all colour capabilities.  Best make the following items optional:
- `state/x`, `state/y` are mandatory only when `cap/color/capabilities` includes `xy`;
- `state/hue`, `state/sat` are mandatory only when `cap/color/capabilities` include `hs`;
- `state/effect` is mandatory only when `cap/color/capabilities` includes `effect`.

Note that either `state/x` and `state/y` or `state/hue` and `state/sat` are mandatory (or the light would be a _Color Temperature Light_), but that might be too complex for validator.

File: devices/generic/subdevices/extended_color_light.json [hunk 1]
```diff
@@ -6,8 +6,6 @@
     "order": 10,
     "uuid": ["$address.ext", "0x01"],
     "items": [
```
Comment (ebaauw) [inline devices/generic/subdevices/extended_color_light.json:8]: `state/reachable` should be mandatory.

Comment (Zehir) [reply devices/generic/subdevices/extended_color_light.json:8]: `state/reachable` is already in the items list

File: devices/generic/subdevices/extended_color_light.json [hunk 2]
```diff
@@ -6,9 +6,15 @@
...
+        "state/alert",
+        "state/colormode",
+        "state/on",
+        "state/reachable"
+    ],
+    "items_optional":[
         "cap/alert/trigger_effect",
         "cap/bri/move_with_onoff",
-        "cap/color/capabilities",
         "cap/color/ct/max",
```
Comment (ebaauw) [inline devices/generic/subdevices/extended_color_light.json:18]: `cap/color/ct/max` and `cap/color/ct/min` are mandatory, as _Extended Color Light_ also has mandatory `state/ct`.

File: devices/generic/subdevices/extended_color_light.json [hunk 3]
```diff
@@ -17,20 +23,18 @@
...
+        "cap/transition_block",
         "config/bri/execute_if_off",
         "config/bri/startup",
         "config/color/ct/startup",
         "config/color/execute_if_off",
         "config/on/startup",
-        "state/alert",
         "state/bri",
-        "state/colormode",
         "state/ct",
```
Comment (ebaauw) [inline devices/generic/subdevices/extended_color_light.json:35]: `state/ct` is mandatory (or it would be a _Color Light_).

**Review by ebaauw — CHANGES_REQUESTED** [review]

**Review by ebaauw — APPROVED** [review]

**Pull request merged**